### PR TITLE
SRV-67 Send campaign data to adPay

### PR DESCRIPTION
**Pull request by PawelPodkalicki**

### CODE REVIEW

File: app/Client/AdPayClient.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+interface AdPayClient
+{
+    public function updateCampaign(array $campaigns): void;
+
+    public function deleteCampaign(array $campaignIds): void;
+
+    public function addEvents(array $events): void;
+
+    public function getPayments(int $timestampFrom, int $timestampTo): array;
+}
```
Comment (c3zi) [inline app/Client/AdPayClient.php]: I'd move this one interface into our Domain :) (src/Demand/Application/Service/AdPay.php)
In the same way, we use AdSelect Client interface For Supply context.
Then we have only our JsonRpc implementation in app/ layer.
Sounds ok?

File: app/Console/Commands/AdPayCampaignExportCommand.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                }
+
+                $mapped = [];
+                $mapped['uuid'] = $campaign->uuid;
+                $mapped['status'] = $campaign->basic_information['status'];
+                $mapped['time_start'] = DateTime::createFromFormat(DATE_ATOM, $campaign->time_start)->getTimestamp();
+                $mapped['time_end'] =
+                    ($campaign->time_end === null)
+                        ? null : DateTime::createFromFormat(DATE_ATOM, $campaign->time_end)
+                        ->getTimestamp();
```
Comment (c3zi) [inline app/Console/Commands/AdPayCampaignExportCommand.php]: Are we sure that `null` means - don't send to AdPay? In adselect we're adding +1 year but in AdPay maybe is risky.
Is AdPay ready to get `null` in the `time_end` property?

**Review by c3zi — CHANGES_REQUESTED** [review]

File: app/Client/AdPayClient.php (hunk 1 above)
Comment (PawelPodkalicki) [reply app/Client/AdPayClient.php]: OK

File: app/Console/Commands/AdPayCampaignExportCommand.php (hunk 2 above)
Comment (PawelPodkalicki) [reply app/Console/Commands/AdPayCampaignExportCommand.php]: Noted.

Comment (PawelPodkalicki) [reply app/Console/Commands/AdPayCampaignExportCommand.php]: Added to question about AdPay API.

**Comment by PawelPodkalicki** [comment]
Please, do not merge. I will add new adPay request format.

**Comment by PawelPodkalicki** [comment]
Format updated. Please review.

**Review by c3zi — APPROVED** [review]

**Pull request merged**